packages/functions_client/lib/src/functions_client.dart

```diff
@@ -52,67 +65,42 @@ class FunctionsClient {
   }) async {
     final bodyStr = body == null ? null : await _isolate.encode(body);
 
-    late final Response response;
     final uri = Uri.parse('$_url/$functionName');
 
     final finalHeaders = <String, String>{
       ..._headers,
       if (headers != null) ...headers
     };
 
-    switch (method) {
-      case HttpMethod.post:
-        response = await (_httpClient?.post ?? http.post)(
-          uri,
-          headers: finalHeaders,
-          body: bodyStr,
-        );
-        break;
-
-      case HttpMethod.get:
-        response = await (_httpClient?.get ?? http.get)(
-          uri,
-          headers: finalHeaders,
-        );
-        break;
-
-      case HttpMethod.put:
-        response = await (_httpClient?.put ?? http.put)(
-          uri,
-          headers: finalHeaders,
-          body: bodyStr,
-        );
-        break;
-
-      case HttpMethod.delete:
-        response = await (_httpClient?.delete ?? http.delete)(
-          uri,
-          headers: finalHeaders,
-        );
-        break;
-
-      case HttpMethod.patch:
-        response = await (_httpClient?.patch ?? http.patch)(
-          uri,
-          headers: finalHeaders,
-          body: bodyStr,
-        );
-        break;
-    }
+    final request = http.Request(method.name, uri);
 
+    finalHeaders.forEach((key, value) {
+      request.headers[key] = value;
+    });
+    if (bodyStr != null) request.body = bodyStr;
+    final response = await (_httpClient?.send(request) ?? request.send());
     final responseType = (response.headers['Content-Type'] ??
```

Comment:
From the [documentation](https://pub.dev/documentation/http/latest/http/BaseResponse/headers.html) it says, that all headers are converted to lowercase. So the first check should never succeed.

Reply:
Hmm, I deleted the `response.headers['Content-Type'] ??` and ran the test, but it seems like the tests are failing when the mock server returns `Content-Type` header in upper case.

Reply:
I still think the documentation is correct, but we just skip the translation when mocking the response object directly. So just keep it.